pkg/gofr/cmd.go

```diff
@@ -1,70 +1,173 @@
 package gofr
 
 import (
+	"fmt"
 	"os"
 	"regexp"
-
-	"gofr.dev/pkg/gofr/container"
+	"strings"
 
 	cmd2 "gofr.dev/pkg/gofr/cmd"
+	"gofr.dev/pkg/gofr/container"
 )
 
 type cmd struct {
 	routes []route
 }
 
 type route struct {
-	pattern string
-	handler Handler
+	pattern     string
+	handler     Handler
+	description string
+	help        string
+	fullPattern string
 }
 
+type Options func(c *route)
+
 type ErrCommandNotFound struct{}
 
 func (e ErrCommandNotFound) Error() string {
 	return "No Command Found!" //nolint:goconst // This error is needed and repetition is in test to check for the exact string.
 }
 
+func (cmd *cmd) Validate(data []string) bool {
+	for _, val := range data {
+		if val != "" {
+			return false
+		}
+	}
+
+	return true
+}
+
 func (cmd *cmd) Run(c *container.Container) {
 	args := os.Args[1:] // First one is command itself
-	command := ""
+	command := []string{}
+	tempCommand := ""
 
-	// Removing all flags and putting everything else as a part of command.
-	// So, unlike native flag package we can put subcommands anywhere
+	showHelp := false
 	for _, a := range args {
 		if a == "" {
-			continue // This takes cares of cases where command has multiple space in between.
+			continue // This takes care of cases where command has multiple spaces in between.
+		}
+
+		if a == "-h" || a == "--help" {
+			showHelp = true
+			continue
 		}
 
 		if a[0] != '-' {
-			command = command + " " + a
+			tempCommand = tempCommand + " " + a
+		} else {
+			command = append(command, tempCommand)
+			tempCommand = a
 		}
 	}
 
-	h := cmd.handler(command)
-	ctx := newContext(&cmd2.Responder{}, cmd2.NewRequest(args), c)
+	if tempCommand != "" {
+		command = append(command, tempCommand)
+	}
 
-	if h == nil {
-		ctx.responder.Respond(nil, ErrCommandNotFound{})
+	if showHelp || len(command) == 0 {
+		cmd.printHelp()
 		return
 	}
 
-	ctx.responder.Respond(h(ctx))
+	ctx := newContext(&cmd2.Responder{}, cmd2.NewRequest(command), c)
+
+	for it, commandVal := range command {
+		if commandVal == "" {
+			continue
+		}
+
+		h := cmd.handler(commandVal)
+
+		if h == nil {
+			ctx.responder.Respond(nil, ErrCommandNotFound{})
+			return
+		}
+
+		ctx.responder.Respond(h(ctx))
+
+		if it != len(command)-1 {
+			ctx.responder.Respond("\n", nil)
+		}
+	}
 }
 
 func (cmd *cmd) handler(path string) Handler {
+	// Trim leading dashes
+	path = strings.TrimPrefix(strings.TrimPrefix(path, "--"), "-")
```

Comment:
You didn't reply to my thread but fixed it somehow

https://github.com/gofr-dev/gofr/pull/661#discussion_r1632321861

I'm unsure why you are not using `strings.Trim(path, "-")`

Especially now you added a comment about removing trailing dashes